src/Symfony/Component/BrowserKit/Tests/ClientTest.php

```diff
@@ -620,6 +620,43 @@ public function testSetServerParameter()
         $this->assertEquals('testua', $client->getServerParameter('HTTP_USER_AGENT'));
     }
 
+    public function testUnSetServerParameter()
+    {
+        $client = new TestClient();
+
+        $this->assertSame('', $client->getServerParameter('HTTP_HOST'));
+        $this->assertSame('Symfony BrowserKit', $client->getServerParameter('HTTP_USER_AGENT'));
+
+        $client->setServerParameter('HTTP_HOST', 'testhost');
+        $client->unSetServerParameter('HTTP_HOST');
+
+        $this->assertSame('', $client->getServerParameter('HTTP_HOST'));
+
+        $client->setServerParameter('HTTP_USER_AGENT', 'testua');
+        $client->unSetServerParameter('HTTP_USER_AGENT');
+
+        $this->assertSame('', $client->getServerParameter('HTTP_USER_AGENT'));
+
+        $client->setServerParameter('HTTP_X-Requested-With', 'XMLHttpRequest');
+        $client->unSetServerParameter('HTTP_X-Requested-With');
+
+        $client->unSetServerParameter('HTTP_X-fefef-With');
+
+        $this->assertSame('', $client->getServerParameter('HTTP_X-Requested-With'));
+
+        $client->request('GET', 'https://www.example.com/https/www.example.com', array(), array(), array(
```

Comment:
Yes :)

I also think it could be simplified to:

```php
    public function testUnSetServerParameter()
     {
         $client = new TestClient();
 
         $client->setServerParameter('HTTP_X-Requested-With', 'XMLHttpRequest');

         $this->assertSame('XMLHttpRequest', $client->getServerParameter('HTTP_X-Requested-With'));

         $client->unSetServerParameter('HTTP_X-Requested-With');
         $client->request('GET', 'https://www.example.com/https/www.example.com');
 
         $this->assertSame('', $client->getServerParameter('HTTP_X-Requested-With'));
         $this->assertArrayNotHasKey('HTTP_X-Requested-With', $client->getInternalRequest()->getServer());
     }
```

WDYT? (I don't see what else it tests)